compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -874,15 +874,18 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         binder: NodeId,
         generic_params: &[GenericParam],
     ) -> &'hir [hir::GenericParam<'hir>] {
-        let mut generic_params: Vec<_> = self
-            .lower_generic_params_mut(generic_params, hir::GenericParamSource::Binder)
-            .collect();
+        let mut params = vec![];
+
         let extra_lifetimes = self.resolver.take_extra_lifetime_params(binder);
         debug!(?extra_lifetimes);
-        generic_params.extend(extra_lifetimes.into_iter().filter_map(|(ident, node_id, res)| {
+        params.extend(extra_lifetimes.into_iter().filter_map(|(ident, node_id, res)| {
```

Comment:
Why does this fix the issue?

Reply:
An invocation of a `﻿def_id` in `extra_lifetimes` occurs during ﻿`lower_generic_params_mut`. This rearrangement ensures that the ﻿`def_id` is not ﻿None.